drivers/scheduler/k8s/k8s.go

```diff
@@ -5134,6 +5144,73 @@ func (k *K8s) createAdmissionRegistrationObjects(
 	return nil, nil
 }
 
+// createVirtualMachineObjects creates the kubevirt VirtualMachines using kubectl apply
+func (k *K8s) createVirtualMachineObjects(
+	spec interface{},
+	ns *corev1.Namespace,
+	app *spec.AppSpec,
+) (interface{}, error) {
+	if obj, ok := spec.(*kubevirtv1.VirtualMachine); ok {
+		// Validating to make sure the desired images are imported by CDI and volumes are ready to be used by Kubevirt VM
+		virtualMachineVolumes := obj.Spec.Template.Spec.Volumes
+		if len(virtualMachineVolumes) > 0 {
+			for _, v := range virtualMachineVolumes {
+				// Validating Volume Source of type PersistentVolumeClaim is ready to be used by Kubevirt VM
+				if v.VolumeSource.PersistentVolumeClaim != nil {
```

Comment:
Does it make sense to make this a separate method - for better and future usability

Reply:
I agree. I will convert this whole block to a function called `WaitForImageImportForVM`